### PR TITLE
CB-11904 CB backend code for CRUD operations on custom catalogs

### PR DESCRIPTION
**Pull request by TheTinkerDad**
Partial implementation in based on the "contract first" design philosophy - right now having the fixed API interfaces finalized and made available for other subsystems is way more important than the polished backend implementation, so implementation of those subsystems can progress without being blocked by this.

### CODE REVIEW

File: freeipa/src/main/java/com/sequenceiq/freeipa/dto/ImageWrapper.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import com.sequenceiq.freeipa.api.model.image.Image;
+
+public class ImageWrapper {
+
+    private Image image;
+
+    private String catalogUrl;
+
+    private String catalonName;
```
Comment (Bajzathd) [inline freeipa/src/main/java/com/sequenceiq/freeipa/dto/ImageWrapper.java:11]: typo

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/image/FreeIpaImageProvider.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    @Value("${freeipa.image.catalog.default.os}")
+    private String defaultOs;
+
+    @Value("${info.app.version:}")
+    private String freeIpaVersion;
+
+    @Override
+    public Optional<ImageWrapper> getImage(ImageSettingsRequest imageSettings, String region, String platform) {
+        String imageId = imageSettings.getId();
+        String catalogUrl = Objects.nonNull(imageSettings.getCatalog()) ? imageSettings.getCatalog() : defaultCatalogUrl;
```
Comment (Bajzathd) [inline freeipa/src/main/java/com/sequenceiq/freeipa/service/image/FreeIpaImageProvider.java:47]: nit: Objects.requireNonNullElse() would be better

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/image/FreeIpaImageProvider.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        if (Objects.nonNull(imageOs) && !imageOs.isEmpty()) {
+            images = filterImages(images, imageOs, platform, region);
+        }
+        if (Objects.nonNull(imageId) && !imageId.isEmpty()) {
+            return images.stream()
+                    .filter(img -> img.getImageSetsByProvider().containsKey(platform) && img.getImageSetsByProvider().get(platform).containsKey(region))
+                    .filter(img -> img.getImageSetsByProvider().get(platform).get(region).equalsIgnoreCase(imageId))
+                    .collect(Collectors.toList());
+        } else {
+            return images.stream().filter(image -> image.getImageSetsByProvider().containsKey(platform))
```
Comment (Bajzathd) [inline freeipa/src/main/java/com/sequenceiq/freeipa/service/image/FreeIpaImageProvider.java:77]: The if filters for region too (not just platform), do we not need that here?

File: freeipa/src/main/java/com/sequenceiq/freeipa/dto/ImageWrapper.java (hunk 1 above)
Comment (pkedvessy) [reply freeipa/src/main/java/com/sequenceiq/freeipa/dto/ImageWrapper.java:11]: True :-)

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/image/FreeIpaImageProvider.java (hunk 2 above)
Comment (pkedvessy) [reply freeipa/src/main/java/com/sequenceiq/freeipa/service/image/FreeIpaImageProvider.java:49]: Copy-paste from ImageService

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/image/FreeIpaImageProvider.java (hunk 3 above)
Comment (pkedvessy) [reply freeipa/src/main/java/com/sequenceiq/freeipa/service/image/FreeIpaImageProvider.java:79]: Copy-paste from ImageService

File: freeipa/src/main/java/com/sequenceiq/freeipa/dto/ImageWrapper.java (hunk 1 above)
Comment (pkedvessy) [reply freeipa/src/main/java/com/sequenceiq/freeipa/dto/ImageWrapper.java]: Fixed

**Review by Bajzathd — APPROVED** [review]

**Pull request merged**